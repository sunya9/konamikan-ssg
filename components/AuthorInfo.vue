<template>
  <div class="columns is-centered">
    <div class="column is-10">
      <div class="box">
        <div class="media">
          <div class="media-left">
            <div class="image is-64x64">
              <nuxt-link :to="`/author/${author.slug}`">
                <img
                  class="is-rounded"
                  :src="author.profile_image"
                  width="128"
                  height="128"
                  loading="lazy"
                />
              </nuxt-link>
            </div>
          </div>
          <div class="media-content">
            <div class="content is-small">
              <div class="title is-6">
                <nuxt-link :to="`/author/${author.slug}`">
                  {{ author.name }}
                </nuxt-link>
              </div>
              <div class="subtitle is-6">
                <small v-if="author.location" class="has-text-grey">
                  {{ author.location }}
                </small>
              </div>
              <p>{{ author.bio }}</p>
              <author-web :author="author" />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'
import { Author } from '@tryghost/content-api'
import AuthorWeb from '~/components/AuthorWeb.vue'

@Component({
  components: { AuthorWeb }
})
export default class AuthorInfo extends Vue {
  @Prop({ type: Object, required: true })
  author!: Author
}
</script>
