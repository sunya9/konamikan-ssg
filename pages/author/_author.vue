<template>
  <div>
    <app-header
      home-button
      :title="author.name"
      :description="author.bio"
      :cover="$resolveUrl(author.cover_image)"
    >
      <div class="container has-text-centered">
        <div class="media">
          <div class="media-left">
            <div class="image is-128x128">
              <img
                class="is-rounded"
                :src="author.profile_image"
                width="256"
                height="256"
                loading="lazy"
              />
            </div>
          </div>
          <div class="media-content">
            <div class="content">
              <h1 class="title is-1 has-text-white">
                {{ author.name }}
              </h1>
              <h2
                v-if="author.location"
                class="subtitle has-text-white-ter is-6"
              >
                {{ author.location }}
              </h2>
              <p class="has-text-white-ter">
                {{ author.bio }}
              </p>
              <author-web
                :author="author"
                color="has-text-white"
                size="is-normal"
              />
            </div>
          </div>
        </div>
      </div>
    </app-header>
    <main class="section">
      <div class="container">
        <posts-grouped-by-year-wrapper :posts="posts" />
      </div>
    </main>
  </div>
</template>
<script lang="ts">
import Vue from 'vue'
import { Component } from 'nuxt-property-decorator'
import { Author, AuthorsObject, PostObject } from '@tryghost/content-api'
import PostsGroupedByYearWrapper from '~/components/PostsGroupedByYearWrapper.vue'
import AppHeader from '~/components/AppHeader.vue'
import AuthorWeb from '~/components/AuthorWeb.vue'
import { reducePostFieldMapper, PostOrPageLight } from '~/util/util'

@Component({
  components: {
    PostsGroupedByYearWrapper,
    AppHeader,
    AuthorWeb
  },
  async asyncData({ $http, route, error }) {
    const [postsWrapper, authorsWrapper] = await Promise.all([
      $http.$get<PostObject>('/resources/posts'),
      $http.$get<AuthorsObject>('/resources/authors')
    ])
    const posts = postsWrapper.posts
    const authors = authorsWrapper.authors
    const author = authors.find((author) => author.slug === route.params.author)
    if (!author) return error({ statusCode: 404 })
    return {
      posts: posts
        .filter((post) =>
          post.authors?.some((postAuthor) => postAuthor.slug === author.slug)
        )
        .map(reducePostFieldMapper),
      author
    }
  }
})
export default class AuthorPage extends Vue {
  author!: Author
  posts!: PostOrPageLight[]
  head() {
    const meta: object[] = [
      {
        hid: 'description',
        property: 'description',
        content: this.author.bio
      },
      ...this.$createBaseMetadata(this.author)
    ]
    return {
      title: this.author.meta_title || this.author.name,
      meta
    }
  }
}
</script>
