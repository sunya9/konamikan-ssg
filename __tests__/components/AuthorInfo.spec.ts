import { mount } from '@vue/test-utils'
import { author } from '@/__tests__/fixtures/author'
import AuthorInfo from '@/components/AuthorInfo.vue'

describe('AuthorInfo', () => {
  it('normal', () => {
    expect(
      mount(AuthorInfo, {
        propsData: {
          author
        }
      })
    ).toMatchSnapshot()
  })
})
