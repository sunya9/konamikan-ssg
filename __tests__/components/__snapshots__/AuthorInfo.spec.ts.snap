// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorInfo normal 1`] = `
<div class="columns is-centered">
  <div class="column is-10">
    <div class="box">
      <div class="media">
        <div class="media-left">
          <div class="image is-64x64"><a><img src="" width="128" height="128" loading="lazy" class="is-rounded"></a></div>
        </div>
        <div class="media-content">
          <div class="content is-small">
            <div class="title is-6"><a>
                Tester
              </a></div>
            <div class="subtitle is-6">
              <!---->
            </div>
            <p></p>
            <div class="level is-mobile">
              <div class="level-left">
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
